pkg/virtctl/adm/logverbosity/logverbosity_test.go

```diff
@@ -0,0 +1,391 @@
+			cmd := clientcmd.NewRepeatableVirtctlCommand("adm", "log-verbosity", argStr)
+			err := cmd()
+			Expect(err).NotTo(Succeed())
+		},
+			Entry("reset and all coexist", "--reset", "--all=3"),
+			Entry("invalid argument (negative verbosity)", "--virt-api=-1"),
+			Entry("invalid argument (character)", "--virt-api=a"),
+			Entry("unknown flag", "--node"),
+			Entry("invalid flag format", "--all", "3"),
+		)
+
+		DescribeTable("should fail handled by error handler", func(output string, args ...string) {
+			commandAndArgs := []string{"adm", "log-verbosity"}
+			commandAndArgs = append(commandAndArgs, args...)
+			_, err := clientcmd.NewRepeatableVirtctlCommandWithOut(commandAndArgs...)()
+			Expect(err).NotTo(Succeed())
+
+			Expect(err).To(MatchError(ContainSubstring(output)))
+		},
+			Entry("show and set mix", "only show or set is allowed", "--virt-handler", "--virt-launcher=3"),
+			Entry("show and reset mix", "only show or set is allowed", "--reset", "--virt-launcher"),
+			Entry("11 or above verbosity", "virt-api: log verbosity must be 0-9", "--virt-api=11"),
+			Entry("one valid verbosity, one invalid verbosity", "virt-handler: log verbosity must be 0-9", "--virt-api=5", "--virt-handler=20"),
+		)
+	})
+
+	When("no logVerbosity field in the KubeVirt CR", func() {
+		BeforeEach(func() {
+			kvInterface.EXPECT().List(gomock.Any()).Return(kvs, nil).AnyTimes()
+			kvInterface.EXPECT().Get(kvs.Items[0].Name, gomock.Any()).Return(&kvs.Items[0], nil).AnyTimes()
+		})
+
+		DescribeTable("show operation", func(output []uint, args ...string) {
+			// should show the logVerbosity for components from the KubeVirt CR
+			// should show the unattended verbosity, so fill them with default verbosity (2)
+			commandAndArgs := []string{"adm", "log-verbosity"}
+			commandAndArgs = append(commandAndArgs, args...)
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(commandAndArgs...)()
+			Expect(err).To(Succeed())
+
+			message := createOutputMessage(output) // create an expected output message
+			Expect(string(bytes)).To(ContainSubstring(*message))
+		},
+			Entry("all components", []uint{2, 2, 2, 2, 2}, "--all"),
+			Entry(
+				"one component (1st component (i.e. virt-api))",
+				[]uint{2, logverbosity.NoFlag, logverbosity.NoFlag, logverbosity.NoFlag, logverbosity.NoFlag},
+				"--virt-api",
+			),
+			Entry(
+				"one component (last component (i.e. virt-operator))",
+				[]uint{logverbosity.NoFlag, logverbosity.NoFlag, logverbosity.NoFlag, logverbosity.NoFlag, 2},
+				"--virt-operator",
+			),
+			Entry(
+				"two components",
+				[]uint{logverbosity.NoFlag, 2, 2, logverbosity.NoFlag, logverbosity.NoFlag},
+				"--virt-controller",
+				"--virt-handler",
+			),
+			Entry("all + one component", []uint{2, 2, 2, 2, 2}, "--all", "--virt-launcher"),
+		)
+
+		Describe("set operation", func() {
+			Context("reset", func() {
+				It("do nothing", func() {
+					// patch = {}
+					// do nothing (do not call Patch method)
```

Comment:
nit: do we need this comment?

Reply:
No. It is clear from the test description. Will remove it.